### PR TITLE
Shapr version 1.0.0

### PR DESCRIPTION
**Pull request by martinju**
This should merge to master what eventually becomes shapr 1.0.0.
All changes up until that point should be made through separate PRs to be merge into the shapr_1.0.0 branch

Required in this PR 
- [x] Add adaptive estimation
- [x] Add verbose
- [x] Get shaprpy to work again
- [x] Get explain_forecast to work again
- [x] Update explain_forecast to the new adaptive version
- [x] Add asymmetric/causal to explain

Other mandatory TODOs
 - [x] Update the NEWS file

Minor TODOs (some might be skipped before release)
- [x] Consider moving compute_sd, n_boot_samps, max_batch_size, min_n_batches, saving_path out of adaptive_arguments and into a new argument list called something like `additional_estimation_args` #404 
- [x] Consider renaming `adaptive_arguments` to `adaptive_args` #404 
- [x] Consider moving keep_samp_for_vS and MSEv_uniform_comb_weights into a input arg called something like output_args #404 
- [x] Consider renaming shapley_reweighting to kernelSHAP_reweighting #404 
- [x] Consider moving saving_path and iter_res_dt to the main output to make it easier to access for users #404 
- [x] add skip_on_os("mac)" for the vaeac tests which fails on mac due to differenences in torch seed (after Lars have verified the results locally on mac)

### CODE REVIEW

File: R/approach_timeseries.R [hunk 1]
```diff
@@ -134,12 +136,12 @@ prepare_data.timeseries <- function(internal, index_features = NULL, ...) {
       names(tmp[[j]]) <- names(tmp[[1]])
     }
 
-    dt_l[[i]] <- rbindlist(tmp, idcol = "id_combination")
-    # dt_l[[i]][, w := 1 / .N, by = id_combination] # IS THIS NECESSARY?
+    dt_l[[i]] <- rbindlist(tmp, idcol = "id_coalition")
+    # dt_l[[i]][, w := 1 / .N, by = id_coalition] # IS THIS NECESSARY?
```
Comment (aredelmeier) [inline R/approach_timeseries.R:140]: Delete

File: R/approach_vaeac.R [hunk 2]
```diff
@@ -185,23 +197,25 @@ setup_approach.vaeac <- function(internal, # add default values for vaeac here.
...
 #' @author Lars Henry Berge Olsen
 prepare_data.vaeac <- function(internal, index_features = NULL, ...) {
+  iter <- length(internal$iter_list)
+
+  n_coalitions <- internal$iter_list[[iter]]$n_coalitions
+  S <- internal$iter_list[[iter]]$S
+
   # If not provided, then set `index_features` to all non trivial coalitions
-  if (is.null(index_features)) index_features <- seq(2, internal$parameters$n_combinations - 1)
+  if (is.null(index_features)) index_features <- seq(2, n_coalitions - 1)
```
Comment (aredelmeier) [inline R/approach_vaeac.R:206]: Nit pick: I don't love if else without the curly brackets

File: R/compute_estimates.R [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+  dt_shapley_sd[, explain_id := .I]
+  setcolorder(dt_shapley_sd, "explain_id")
+
+
+  internal$iter_list[[iter]]$dt_shapley_est <- dt_shapley_est
+  internal$iter_list[[iter]]$dt_shapley_sd <- dt_shapley_sd
+  internal$iter_list[[iter]]$vS_list <- vS_list
+  internal$iter_list[[iter]]$dt_vS <- processed_vS_list$dt_vS
+
+  # internal$timing$shapley_computation <- Sys.time()
```
Comment (aredelmeier) [inline R/compute_estimates.R:64]: Delete

File: R/compute_vS.R [hunk 4]
```diff
@@ -1,23 +1,49 @@
...
 compute_vS <- function(internal, model, predict_model, method = "future") {
-  S_batch <- internal$objects$S_batch
+  iter <- length(internal$iter_list)
+
+  S_batch <- internal$iter_list[[iter]]$S_batch
+
+  verbose <- internal$parameters$verbose
+  approach <- internal$parameters$approach
+
+  # verbose
```
Comment (aredelmeier) [inline R/compute_vS.R:20]: Necessary?

File: R/documentation.R [hunk 5]
```diff
@@ -46,7 +48,7 @@ default_doc_explain <- function(internal, index_features) {
 #' @description
 #' This helper function displays the specific arguments applicable to the different
 #' approaches. Note that when calling [shapr::explain()] from Python, the parameters
-#' are renamed from the form `approach.parameter_name` to `approach_parameter_name`.
+#' are renamed from the from `approach.parameter_name` to `approach_parameter_name`.
```
Comment (aredelmeier) [inline R/documentation.R:51]: Remove 'from the'

File: R/finalize_explanation.R [hunk 6]
```diff
@@ -262,16 +207,72 @@ compute_MSEv_eval_crit <- function(internal,
...
+
+  processed_vS_list <- postprocess_vS_list(
+    vS_list = vS_list,
+    internal = internal
+  )
+
+  # Extract the predictions we are explaining
+  p <- get_p(processed_vS_list$dt_vS, internal)
+
+  # internal$timing$postprocessing <- Sys.time()
```
Comment (aredelmeier) [inline R/finalize_explanation.R:244]: Remove

File: R/finalize_explanation.R [hunk 7]
```diff
@@ -262,16 +207,72 @@ compute_MSEv_eval_crit <- function(internal,
...
+
+  # Compute the Shapley values
+  dt_shapley <- compute_shapley_new(internal, processed_vS_list$dt_vS)
+
+  # internal$timing$shapley_computation <- Sys.time()
```
Comment (aredelmeier) [inline R/finalize_explanation.R:249]: Remove

File: R/save_results.R [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+  filename <- basename(saving_path)
+  dirname <- dirname(saving_path)
+  filename_copy <- paste0("new_", filename)
+  saving_path_copy <- file.path(dirname, filename_copy)
+
+  # Save the results to a new location, then delete old and rename for safe code interruption
+
+  # Saving parameters and iter_list
+  saveRDS(internal[c("parameters", "iter_list")], saving_path_copy)
+  if (file.exists(saving_path)) file.remove(saving_path)
```
Comment (aredelmeier) [inline R/save_results.R:20]: Don't like if without curly bracket

File: R/explain.R [hunk 9]
```diff
@@ -345,55 +410,104 @@ explain <- function(model,
...
+    # Save intermediate results
+    save_results(internal)
+
+    # Preparing parameters for next iteration (does not do anything if already converged)
+    internal <- prepare_next_iteration(internal)
+
+    # Printing iteration information
+    print_iter(internal)
+
+    ### Setting globals for to simplify the loop
```
Comment (aredelmeier) [inline R/explain.R:462]: Remove two #

File: R/approach_vaeac.R (hunk 2 above)
Comment (martinju) [reply R/approach_vaeac.R:206]: I generally agree, but these very short ones are OK I think. Keep.

**Comment by martinju** [comment]
@aredelmeier Not sure if you consider yourself done with spell checks etc, but if you are going to do more, you can actually create a separate PR which is proposed merged into the shapr-1.0.0 branch (this one). 
That's more efficient, and we can keep this PR a bit cleaner :-)

**Pull request merged**